tests/integ/modin/index/test_index_methods.py

```diff
@@ -419,3 +420,41 @@ def test_index_identical():
     i3 = pd.Index([("a", "a"), ("a", "b"), ("b", "a")])
     i4 = pd.Index([("a", "a"), ("a", "b"), ("b", "a")], tupleize_cols=False)
     assert not i3.identical(i4)
+
+
+def test_df_set_index_and_reset_index():
+    native_df = native_pd.DataFrame(
+        {"A": [1, 2, 3], "B": [4, 5, 6]}, index=["A", "B", "C"]
+    )
+    snow_df = pd.DataFrame(native_df)
+    native_idx = native_pd.Index([11, 22, 33])
+    snow_idx = pd.Index(native_idx)
+
+    # Test that df.index = new_index works with lazy index.
+    with SqlCounter(query_count=3):
```

Comment:
Is it possible to reduct to single query?

Reply:
Do we understand the reason for the three queries?

Reply:
There's three queries because:
- 2 queries from `set_index` in the query compiler - one from `self.get_axis_len(axis=0)` and another from `key.get_axis_len(0)`
- 1 query to perform `to_pandas` for comparison

I'm going to see if I can get the number of queries down but right now I'm not sure how to reduce the query count without eliminating the length checks.

Reply:
Could we lazily fail if the lengths aren't correct?

Reply:
`if key.get_axis_len(0) != self_num_rows:`
at least we can drop 1 query by check the two lengths into a single query.